client/src/components/ProfileEdit/index.js

```diff
@@ -131,47 +94,47 @@ const ProfileEdit = function () {
   } = useForm(initialForm);
 
   useEffect(() => {
-    fetch(`/profile/${user.profile}`)
-      .then((res) => res.json())
-      .then((profile) =>
-        setValues((values) => ({
-          ...values,
-          ...profile,
-          gender:
-            profile.gender === "non-binary"
-              ? "Non-Binary"
-              : capitaleachword(profile.gender),
-          birthDate: new Date(profile.birthDate),
-          email: user.email,
-        }))
-      );
-  }, [user.email, user.profile, setValues]);
+    setValues((values) => ({
+      ...values,
+      ...profile,
+      gender: genderCheck(profile.gender),
+      birthDate: new Date(profile.birthDate),
+      email: user.email,
+    }));
+  }, [profile, user, setValues]);
 
-  const handlePhone = (value) => {
-    if (!/\+1 \(\d{3}\) \d{3}-\d{4}/.test(value)) {
-      setErrors({ ...errors, phone: "Phone number is invalid" });
+  const regExHandler = (regEx, errMsg, prop) => (e) => {
+    const value = e.target.value;
+    if (!regEx.test(value)) {
+      setErrors({ ...errors, [prop]: errMsg });
       setDisableSubmit(true);
     } else {
-      setErrors({ ...errors, phone: "" });
+      setErrors({ ...errors, [prop]: "" });
       setDisableSubmit(false);
     }
 
-    setValues({ ...values, phone: value });
+    setValues({ ...values, [prop]: value });
   };
 
-  const handleEmail = (e) => {
-    const text = e.currentTarget.value;
-    if (!/.+@.+..+/.test(text)) {
-      setErrors({ ...errors, email: "Email is not valid" });
+  const handlePhone = (value) => {
+    if (!/\+1 \(\d{3}\) \d{3}-\d{4}/.test(value)) {
+      setErrors({ ...errors, phone: "Phone number is invalid" });
       setDisableSubmit(true);
     } else {
-      setErrors({ ...errors, email: "" });
+      setErrors({ ...errors, phone: "" });
       setDisableSubmit(false);
     }
 
-    setValues({ ...values, email: text });
+    setValues({ ...values, phone: value });
   };
 
+  const handleEmail = regExHandler(/.+@.+..+/, "Email is not valid", "email");
```

Comment:
This regex evaluator misses a few cases, i.e. tony@test passes the validation. There are some more robust regex checks for email online that we can refer from